specs/howto.md

```diff
@@ -0,0 +1,94 @@
+  help reducing the amount of information exchanged to the one you need and not
+  more than that. This may apply also just to a part of your application. I.e.
+  while you use in your Context Broker the whole data model, to other services
+  you may dispatch only the needed part.
+* Use metadata only in case of need, and follow the assumptions provided in the
+  data model. For example, in case your application is leveraging the default
+  unit defined in a data model (e.g. meters), avoid attaching metadata to
+  annotate the unit. In case your application is using a different unit,
+  do annotate it, it will simplify data exchange with other systems in case of
+  need.  
+* Stick to the data model, the data model are meant to be an effort to
+  facilitate interoperability within the community. If you change the meaning
+  or the data type attached to an attribute, this will broke interoperability.
+  Of course it may be the case that the model is not covering your scenario,
+  or it contains errors. Open a ticket in the repository, and let the community
+  discuss on the issue!
+* While not generally advised, you can add additional attributes in your
+  specific application for the purpose of your application. Be aware that of
+  course this means that such attributes may not be correctly used by other
+  applications leveraging the same data model. We recommend, if the attribute
+  you want to add is of general adoption and not specific to your case, to
+  propose it to become officially part of the data model. You can open an issue
+  or a pull request for that.
+* In several cases the data model proposed are codification of existing data
+  models. You can find the references in each data model specification, but
+  for the sake of maintainability, on purpose we don't report the full
+  specification. This means that to gather further details, you need to access
+  the original source. For example, for the location, we use GeoJSON. You
+  can find the exact semantics of GeoJSON schema in the related standard.
+* Be aware that [JSON Schema](http://json-schema.org/) provided covers only the
+  the so called *keyValue* representation of NGSI v2 context data.
+* Be aware that at the time being none of the official FIWARE Core Enablers
+  enforce schema validation. Thus, in case you want to be sure your application
+  data validates correctly against a given data model, this is up to you.
+
+## Going trough a data model
+
+Each data model is programmatically defined using a
+[JSON Schema](http://json-schema.org/), as early mentioned, the JSON schema
+covers only the so called  *keyValue* representation of NGSI v2 context data.
+Thus the JSON Schema does not cover the *normalised* representation of  NGSI v2.
+
+To clarify the difference among the two representation here is a small example:
+
+
+
+For each data model we provide a set of examples (in some cases also using
+the Normalised representation of  NGSI v2), such examples are all validated
```

Comment:
not needed empty space before NGSI and it should be NGSIv2

Reply:
@jmcanterafonseca - would disagree with you here. **NGSI v2** with space is more popular than **NGSIv2**, easier to read and easier to search for.

Reply:
let me know guys :)

Reply:
agree with Jason